packages/runtime/datastore/src/channelContext.ts

```diff
@@ -83,3 +83,20 @@ export function summarizeChannel(
     addBlobToSummary(summarizeResult, attributesBlobKey, JSON.stringify(channel.attributes));
     return summarizeResult;
 }
+
+export async function summarizeChannelAsync(
+    channel: IChannel,
+    fullTree: boolean = false,
+    trackState: boolean = false,
+): Promise<ISummaryTreeWithStats> {
+    let summarizeResult: ISummaryTreeWithStats;
+    if (channel.summarizeAsync !== undefined) {
```

Comment:
Can you please add comment in this format?
// back-compat: added in version X to support newly added IChannel.summarizeAsync()

Reply:
If that's the only place where we call into IChannel to summarize, then I think we can do rename here to be consistent with other layers. I.e. we can make summarize() be async and just do
```typesript
summarizeResult = await channel.summarize(fullTree, trackState);
```

If it's old version of code, await of non-promise will return value just fine.

That said, I do not believe we allow mixing versions between data store runtime and DDS, so no back-compat should be necessary (disregard my earlier comments on that front).

The only other place where rename will touch is sync flow